http2_general/test_h2_stream_priority.py

```diff
@@ -0,0 +1,691 @@
+        """
+    }
+
+    def test_stream_priority_stress(self):
+        client, server = self.setup_test_priority()
+        self.run_test_tso_gro_gso_def(
+            client,
+            server,
+            self._test_stream_priority_stress,
+            DEFAULT_MTU,
+        )
+
+    def _test_stream_priority_stress(self, client, server):
+        first_level_streams = []
+        for weight in range(1, 257):
+            first_level_streams.append(client.stream_id)
+            client.make_request(
+                self.post_request,
+                end_stream=True,
+                priority_weight=weight,
+                priority_depends_on=0,
+                priority_exclusive=False,
+            )
+
+        weight = 1
+        for stream_id in first_level_streams:
+            client.make_request(
+                self.post_request,
+                end_stream=True,
+                priority_weight=weight,
+                priority_depends_on=stream_id,
+                priority_exclusive=False,
+            )
+            client.make_request(
+                self.post_request,
+                end_stream=True,
+                priority_weight=257 - weight,
+                priority_depends_on=stream_id,
+                priority_exclusive=False,
+            )
+            weight = weight + 1
+
+        self.wait_for_responses(client, timeout=30)
+        self.check_response_sequence(client, 256 + 256 * 2)
+
+
+class TestMaxConcurrentStreams(TestPriorityBase, NetWorker):
+    tempesta = {
+        "config": """
+            listen 443 proto=h2;
+            max_concurrent_streams 10;
+            srv_group default {
+                server ${server_ip}:8000;
+            }
+            vhost good {
+                proxy_pass default;
+            }
+            tls_certificate ${tempesta_workdir}/tempesta.crt;
+            tls_certificate_key ${tempesta_workdir}/tempesta.key;
+            tls_match_any_server_name;
+
+            block_action attack reply;
+            block_action error reply;
+            http_chain {
+                host == "bad.com"   -> block;
+                                    -> good;
+            }
+        """
+    }
+
+    def test_max_concurent_stream_exceed_by_stream(self):
+        """
+        If creation of new stream leads to exceeding of max_concurrent_streams, we should reset
+        this stream. Check that all streams, which creation leads to exceedion of max_concurrent_streams
+        will be reset and all previos streams will be finished successfully.
+        """
+        client, server = self.setup_test_priority()
+
+        prev_stream_id = 0
+        self.assertTrue(client.h2_connection.remote_settings.max_concurrent_streams == 10)
+        for i in range(0, client.h2_connection.remote_settings.max_concurrent_streams):
+            stream_id = 2 * client.h2_connection.remote_settings.max_concurrent_streams + 2 * i + 1
+            client.send_bytes(
+                PriorityFrame(
```

Comment:
The PRIORITY frame does not open stream. We have 10 Idle streams here (or not?). These stream may not be open, but Temepsta dropped new streams. Maybe Tempesta should not count Idle streams?

Reply:
WE discuss it with Alexander and consider that we should limit idle streams also since we allocate memory for such streams. But we increase default limit of streams up to 200